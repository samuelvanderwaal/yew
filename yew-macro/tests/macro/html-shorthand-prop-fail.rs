use yew::prelude::*;

fn compile_fail() {
    let text = "placeholder-text";
    let length = "20";
    let link = "https://yew.rs/docs/";
    html! { <input placeholder= />};
    html! { <input placeholder=text maxlength= />};
    html! { <input placeholder= maxlength=length />};
    html! { <a href= >{"Yew"}</a>};
    html! { <a href=link target= >{"Yew"}</a>};
}