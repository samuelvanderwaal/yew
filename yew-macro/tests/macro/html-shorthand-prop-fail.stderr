error[E0425]: cannot find value `placeholder` in this scope
 --> $DIR/html-shorthand-prop-fail.rs:7:20
  |
7 |     html! { <input placeholder= />};
  |                    ^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `maxlength` in this scope
 --> $DIR/html-shorthand-prop-fail.rs:8:37
  |
8 |     html! { <input placeholder=text maxlength= />};
  |                                     ^^^^^^^^^ help: a local variable with a similar name exists: `length`

error[E0425]: cannot find value `maxlength` in this scope
 --> $DIR/html-shorthand-prop-fail.rs:9:33
  |
9 |     html! { <input placeholder= maxlength=length />};
  |                                 ^^^^^^^^^ help: a local variable with a similar name exists: `length`

error[E0425]: cannot find value `href` in this scope
  --> $DIR/html-shorthand-prop-fail.rs:10:16
   |
10 |     html! { <a href= >{"Yew"}</a>};
   |                ^^^^ not found in this scope

error[E0425]: cannot find value `target` in this scope
  --> $DIR/html-shorthand-prop-fail.rs:11:26
   |
11 |     html! { <a href=link target= >{"Yew"}</a>};
   |                          ^^^^^^ not found in this scope

error[E0601]: `main` function not found in crate `$CRATE`
  --> $DIR/html-shorthand-prop-fail.rs:1:1
   |
1  | / use yew::prelude::*;
2  | |
3  | | fn compile_fail() {
4  | |     let text = "placeholder-text";
...  |
11 | |     html! { <a href=link target= >{"Yew"}</a>};
12 | | }
   | |_^ consider adding a `main` function to `$DIR/tests/macro/html-shorthand-prop-fail.rs`
